src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.ShopService;
+
+public class PurchaseOperation implements OperationHandler {
+    private ShopService shopService;
+
+    public PurchaseOperation(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int transactionQuantity = transaction.getQuantity();
+        int currentQuantity = shopService.getQuantity(fruit);
+        if (transaction.getOperation() == FruitTransaction.Operation.PURCHASE) {
+            if (transactionQuantity > currentQuantity) {
+                throw new RuntimeException("Not enough " + fruit + " in stock for purchase");
```

Comment:
Throwing a `RuntimeException` for insufficient stock might not be the best approach as it can terminate the application abruptly. Consider using a custom exception or handling the error more gracefully to provide better feedback to the user.